spec/stack_master/sparkle_formation/compile_time/allowed_pattern_validator_spec.rb

```diff
@@ -0,0 +1,67 @@
+require_relative '../../../../lib/stack_master/sparkle_formation/compile_time/allowed_pattern_validator'
+
+RSpec.describe StackMaster::SparkleFormation::CompileTime::AllowedPatternValidator do
+
+  describe '#validate' do
+
+    let(:name) {'name'}
+
+    scenarios = [
+        {definition: {type: :string, allowed_pattern: '^a'}, parameter: 'a', valid: true},
+        {definition: {type: :string, allowed_pattern: '^a'}, parameter: ['a'], valid: true},
+        {definition: {type: :string, allowed_pattern: '^a'}, parameter: 'b', valid: false, error: ['b']},
+        {definition: {type: :string, allowed_pattern: '^a'}, parameter: ['b'], valid: false, error: ['b']},
+
+        {definition: {type: :string, allowed_pattern: '^a', default: 'a'}, parameter: nil, valid: true},
+
+        {definition: {type: :string, allowed_pattern: '^a', multiple: true}, parameter: 'a,ab', valid: true},
+        {definition: {type: :string, allowed_pattern: '^a', multiple: true}, parameter: 'a,,ab', valid: false, error: ['']},
+        {definition: {type: :string, allowed_pattern: '^a', multiple: true}, parameter: 'a,, b', valid: false, error: ['', 'b']},
+
+        {definition: {type: :string, allowed_pattern: '^a', multiple: true, default: 'a,a'}, parameter: nil, valid: true},
+
+        {definition: {type: :number, allowed_pattern: '^1'}, parameter: 1, valid: true},
+        {definition: {type: :number, allowed_pattern: '^1'}, parameter: '1', valid: true},
+        {definition: {type: :number, allowed_pattern: '^1'}, parameter: [1], valid: true},
+        {definition: {type: :number, allowed_pattern: '^1'}, parameter: ['1'], valid: true},
+        {definition: {type: :number, allowed_pattern: '^1'}, parameter: 2, valid: false, error: [2]},
+        {definition: {type: :number, allowed_pattern: '^1'}, parameter: '2', valid: false, error: ['2']},
+
+        {definition: {type: :number, allowed_pattern: '^1', default: '1'}, parameter: nil, valid: true},
+    ]
+
+    subject {described_class.new(name, definition, parameter).tap {|validator| validator.validate}}
+
+    scenarios.each do |scenario|
```

Comment:
Shared examples are a much better way of implementing tests that re-use tests like this. They give a much nicer output when one fails and implement a lot of this for you. https://relishapp.com/rspec/rspec-core/docs/example-groups/shared-examples

Reply:
I'm not convinced that adds anything to readability of the test particularly if it's in another file. 
They are also not reusing the same examples between tests.

Reply:
You can keep shared examples in the same test file and that's what I'd encourage if they are not re-used across specs.

They provide more readable _output_ mainly, giving you what exact example failed. Without them you are left scratching your head about which specific example failed. They decorate it enough to know where to start looking.

Reply:
@shivaman Why not give it a try and see which technique is easier to understand.

Reply:
As discussed, shared_examples doesnt make tests much more readable in this context, and we are not reducing any lines of code. So we will leave the specs as is for the moment